### PR TITLE
bug-report: skip coredump and netstat commands for distroless installation

### PR DESCRIPTION
**Pull request by Kaushal28**
**Please provide a description of this PR:**

Issue: #37235 

Distroless containers do not have utilities like `netstat` and `find`. So commands ran to gather these information for bug report were failing.

This PR determines whether the installation is distroless or not. If it is distroless, it'll skip execution of those commands.

### CODE REVIEW

File: tools/bug-report/pkg/bugreport/bugreport.go [hunk 1]
```diff
@@ -56,6 +60,12 @@ var (
 	bugReportDefaultIstioNamespace = "istio-system"
 	bugReportDefaultInclude        = []string{""}
 	bugReportDefaultExclude        = []string{strings.Join(sets.SortedList(inject.IgnoredNamespaces), ",")}
+
+	istioOperatorGVR = schema.GroupVersionResource{
```
Comment (hanxiaop) [inline tools/bug-report/pkg/bugreport/bugreport.go:64]: I think we've already have this GVR defined in some places.

File: tools/bug-report/pkg/bugreport/bugreport.go [hunk 2]
```diff
@@ -198,6 +217,31 @@ func runBugReportCommand(_ *cobra.Command, logOpts *log.Options) error {
...
 
+func isDistroless(client kube.CLIClient, brConfig *config.BugReportConfig, ctx context.Context) (bool, error) {
+	iops, err := client.Dynamic().Resource(istioOperatorGVR).
+		Namespace(brConfig.IstioNamespace).
+		List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return false, fmt.Errorf("error while retrieving to IstioOperator CR - %v", err)
+	}
+
+	// TODO: WIP: Which iop to consider in case of multiple?
```
Comment (hanxiaop) [inline tools/bug-report/pkg/bugreport/bugreport.go:228]: I think if we have multiple control planes, we need to associate the proxy container with the control plane to determine if it's distroless?

File: tools/bug-report/pkg/bugreport/bugreport.go (hunk 1 above)
Comment (Kaushal28) [reply tools/bug-report/pkg/bugreport/bugreport.go]: It was defined at multiple places previously as well. Moved it to a common place and re-using it now.

File: tools/bug-report/pkg/bugreport/bugreport.go (hunk 2 above)
Comment (Kaushal28) [reply tools/bug-report/pkg/bugreport/bugreport.go]: Added logic to get the effective IOP using `istiod` deployment. I am not sure it is the best possible approach to determine the effectiev IOP for the installation. Suggestions/Feedback would be really helpful here. Thanks.

File: tools/bug-report/pkg/bugreport/bugreport.go [hunk 3]
```diff
@@ -198,6 +212,43 @@ func runBugReportCommand(_ *cobra.Command, logOpts *log.Options) error {
 	return nil
 }
 
+func isDistroless(ctx context.Context, client kube.CLIClient, clientset *kubernetes.Clientset, brConfig *config.BugReportConfig) (bool, error) {
+	// Get istiod deployment
```
Comment (Kaushal28) [inline tools/bug-report/pkg/bugreport/bugreport.go:216]: Added logic to get the effective IOP from multiple available IOPs using `istiod` deployment. I am not sure if it is the best possible approach to determine the effectiev IOP for the installation. Suggestions/Feedback would be really helpful here. Thanks.

**Pull request closed without merging**